File: JS/2.1-Sorting - 1/SelectionSort.cpp
#include<bits/stdc++.h>
using namespace std;
void SelectionSort(int arr[], int N) {
    for(int i=0; i<N-1; i++) {
        int mini=i;
        for(int j=i+1; j<N; j++) {
             if(arr[j]<arr[mini]) mini=j;
        }
          // here we can one extra check that if arr[i]!=arr[min] then only go into swap block else go next.
             int temp=arr[i];
             arr[i]=arr[mini];
             arr[mini]=temp; 
    }
}

// K passes of Selection Sort
void SelectionSort(int arr[], int N, int K) {
    for(int i=0; i<K; i++) {
        int mini=i;
        for(int j=i+1; j<N; j++) {
             if(arr[j]<arr[mini]) mini=j;
        }
          // here we can one extra check that if arr[i]!=arr[min] then only go into swap block else go next.
             int temp=arr[i];
             arr[i]=arr[mini];
             arr[mini]=temp; 
    }
}

File: JS/2.1-Sorting - 1/SelectionSort.js
let arr = [-100, 2, 3, 400, 5, 6, 7, 8, 9]

function InsertionSort(arr) {
  let N = arr.length
  // exactly N-1 passes required to sort
  for (let i = 0; i < N - 1; i++) {
    let mini = i
    for (let j = i + 1; j < N; j++) {
      if (arr[j] < arr[mini]) {
        mini = j
      }
    }
    // swaping logic
    if (i !== mini) {
      ;[arr[i], arr[mini]] = [arr[mini], arr[i]]
    }
  }
  return arr
}

console.log(InsertionSort(arr))
